compile.go

```diff
@@ -420,14 +420,14 @@ func (t *TableObject) Equal(rhs values.Value) bool {
 	if t.Len() != r.Len() {
 		return false
 	}
-	for _, k := range t.keys() {
-		val1, ok1 := t.Get(k)
-		val2, ok2 := r.Get(k)
-		if !ok1 || !ok2 || !val1.Equal(val2) {
-			return false
-		}
-	}
-	return true
+	var isEqual = true
+	// Range over both TableObjects and
+	// compare their properties for equality
+	t.Range(func(k string, v values.Value) {
+		w, ok := r.Get(k)
+		isEqual = isEqual && ok && v.Equal(w)
+	})
+	return isEqual
```

Comment:
If `r` has a key that `t` does not, then they will still compare as equal, right?  Seems like you want to also require that `r` and `t` have the same number of keys.